main.js

```diff
@@ -35,45 +42,66 @@ function show(event) {
 }
 
 function getCooking() {
+  showRecipe();
+  getRecipe();
+}
+
+function showRecipe() {
   event.preventDefault();
   hide(cookpotImage);
   hide(favsList);
+  hide(deleteFavButton);
   hide(backButton);
   show(youShouldMake);
   show(outputRecipe);
   show(favButton);
   show(clearButton);
-
-  getRecipe();
 }
 
 function getRecipe() {
-  if (sideOption) {
+  if (sideOption.checked) {
     currentRecipe = sidesRecipes[getRandomIndex(sidesRecipes)];
   }
-  if (mainDishOption) {
+  if (mainDishOption.checked) {
     currentRecipe = mainDishRecipes[getRandomIndex(mainDishRecipes)];
   }
-  if (dessertOption) {
+  if (dessertOption.checked) {
     currentRecipe = dessertRecipes[getRandomIndex(dessertRecipes)];
   }
 
   outputRecipe.innerText = `${currentRecipe}`;
-  // if (entireMealOption) {
+  // if (entireMealOption.checked) {
   //   outputRecipe.innerText = `${mainDishRecipes[getRandomIndex(mainDishRecipes)]} with a side of ${sidesRecipes[getRandomIndex(sidesRecipes)]} and ${dessertRecipes[getRandomIndex(dessertRecipes)]} for dessert!`;
   // }
 }
 
 function addFav() {
   favRecipes.push(currentRecipe);
-
-  var unorderedList = document.querySelector('.favs-list');
-  var listItem = document.createElement('li');
+  // favsList.innerText = '';
 
   for (var i = 0; i < favRecipes.length; i++) {
-    listItem.innerHTML = `${favRecipes[i]}`;
+    // var text = document.createTextNode(favRecipes[i]);
+    checkbox.type = 'checkbox';
+    checkbox.name = 'delete';
+    checkbox.id = 'delete';
+    checkbox.value = i;
+
     unorderedList.appendChild(listItem);
+    listItem.appendChild(checkbox);
+    listItem.appendChild(label);
+
+    // label.appendChild(text);
+    label.innerHTML = `${favRecipes[i]}`;
+  }
+
+  showFavorites();
+}
+
+function deleteFav() {
+  if (checkbox.value > 0) {
```

Comment:
Have been attempting to store the favRecipes index in the value of the checkbox, in case the user wants to delete that recipe from the array.  Used an if statement to make sure we weren't attempting to access the array index of -1.